web/permissions/permissions.go

```diff
@@ -98,7 +107,9 @@ func extract(c echo.Context) (*permissions.Claims, *permissions.Set, error) {
 	}
 
 	var pset permissions.Set
-	if pset, err = claims.PermissionsSet(); err != nil {
+	if claims.Audience == permissions.RegistrationTokenAudience {
+		pset = registerTokenPermissions
```

Comment:
I think there is a confusion between the registration token used in OAuth 2 dynamic client registration/management protocol, and the registerToken used for the onboarding. The registerToken for onboarding is not a JWT, and the registration token for OAuth 2 should not give the permissions to read/write the instance!